Foundation/NSArray.swift

```diff
@@ -27,18 +27,37 @@ open class NSArray : NSObject, NSCopying, NSMutableCopying, NSSecureCoding, NSCo
         return _SwiftValue.fetch(nonOptional: _storage[index])
     }
     
-    public convenience override init() {
-        self.init(objects: [], count:0)
+    public override init() {
+        _storage.reserveCapacity(0)
     }
-    
-    public required init(objects: UnsafePointer<AnyObject>!, count cnt: Int) {
+
+    public required init(objects: UnsafePointer<AnyObject>?, count cnt: Int) {
+        precondition(cnt >= 0)
+        precondition(cnt == 0 || objects != nil)
+
+        _storage.reserveCapacity(cnt)
+        for idx in 0..<cnt {
+            _storage.append(objects![idx])
+        }
+    }
+
+    public convenience init(objects: UnsafePointer<AnyObject>, count cnt: Int) {
+        self.init()
         _storage.reserveCapacity(cnt)
         for idx in 0..<cnt {
             _storage.append(objects[idx])
         }
     }
-    
-    required public convenience init(arrayLiteral elements: Any...) {
+
+    convenience init(objects elements: AnyObject...) {
+        self.init(objects: elements, count: elements.count)
+    }
+
+    convenience init(object: AnyObject) {
```

Comment:
This one is [documented](https://developer.apple.com/documentation/foundation/nsarray/1411981-init) as `convenience init(object anObject: Any)`. Not a big deal though.

Reply:
I think I saw this one and wasn't sure if NSArray was only supposed to hold references (`AnyObject`) and not all types. I will fix that.

Reply:
I removed this one, there was an already existing `convenience init(object anObject: Any)`